### ISSUE TITLE
Linear scale .copy().nice() modifies original scale domain

### ISSUE THREAD

**Issue by dandelany** [mbostock/d3#2772]
It seems that running `.nice()` on a copy of a linear scale created with `.copy()` modifies not only the copy's domain, but the original's domain as well:

```
var x = d3.scale.linear().domain([1.3, 8.7])
console.log(x.domain); // logs [1.3, 8.7] as expected
var xCopy = x.copy();
xCopy.nice();
console.log(xCopy.domain()); // logs [1, 9] as expected
console.log(x.domain()); // ** unexpectedly logs [1, 9] **
```

**Comment by mbostock** [comment]
This has been fixed in 4.0; see [d3-scale](https://github.com/d3/d3-scale). I probably won’t back-port the fix to 3.x, sorry!

**Comment by dandelany** [comment]
Yep I noticed it was fixed there also - will probably upgrade my app to use the new `d3-scale` soon, just figured I'd call this out. Thanks for taking a look!
